core/v2.go

```diff
@@ -698,6 +699,84 @@ func (core *Core) v2API() *route.Router {
 		/* FIXME */
 		r.Fail(route.Errorf(501, nil, "%s: not implemented", r))
 	})
+
+	r.Dispatch("POST /v2/auth/login", func(r *route.Request) {
+		auth := struct {
+			Username string
+			Password string
+		}{}
+
+		if !r.Payload(&auth) { //Payload reports its own errors
+			return
+		}
+
+		if auth.Username == "" {
+			r.Fail(route.Errorf(http.StatusForbidden, nil, "no username given"))
+			return
+		}
+
+		if auth.Password == "" {
+			r.Fail(route.Errorf(http.StatusForbidden, nil, "no password given"))
+		}
+
+		user, err := core.DB.GetUser(auth.Username, "local")
+		if err != nil {
+			r.Fail(route.Oops(err, "An unknown error occurred when authenticating local user '%s'", auth.Username))
+			return
+		}
+
+		if user == nil || !user.Authenticate(auth.Password) {
+			r.Fail(route.Errorf(http.StatusForbidden, nil, "Incorrect username or password"))
+			return
+		}
+
+		session, err := core.createSession(user)
+		if err != nil {
+			r.Fail(route.Oops(err, "An unknown error occurred when authenticating local user '%s'", auth.Username))
+			return
+		}
+
+		r.SetCookie(SessionCookie(session.UUID.String(), true))
+		r.SetHeader("X-Shield-Session", session.UUID.String())
+
+		r.Success("Logged in as user %s", auth.Username)
+	})
+
+	r.Dispatch("GET /v2/auth/logout", func(r *route.Request) {
+		sessionID := getSessionID(r.Req)
+		if sessionID == "" {
+			//I guess we're okay with not getting a session to logout?...
+			r.Success("No user to logout")
+		}
+
+		id := uuid.Parse(sessionID)
+		if id == nil {
+			r.Fail(route.Bad(nil, "Invalid session ID received"))
```

Comment:
We should probably formulate an `error` object to pass to `Bad()`, and make the FE error message more generic, like "Unable to log out"